Fixed moved header file sha.h

kvoit opened the pull request:
See https://github.com/espressif/esp-idf/issues/3245

grostim commented:
what prevent this PR to be merged ?

kvoit commented:
either the fact that this repository clearly isn't maintained anymore, or the fact that this actually breaks compatibility with older IDF versions ... by now, I wouldn't merge it myself in the current state, but haven't brought myself to write a clean version using ESP_IDF_VERSION_MAJOR/ESP_IDF_VERSION_MINOR defines to determine what to include.

gbetous commented:
Hi there,

Very interested in this project, I wonder if there is a kind of "official fork" still maintained where those MR have been accepted ?

kvoit commented:
I have just implemented my above mentioned clean solution for both versions in my fork.
https://github.com/kvoit/RemoteDebug.git

I have also taken care of the deprecation warning of esp/sha.h and replaced it with sha/sha_parallel_engine.h. This compiles, but I don't have the hardware here to finally test.
